### PR TITLE
DATAUP-389 - throw an error if a parent job id is specified in batch runs

### PR DESCRIPTION
**Pull request by MrCreosote**
# Description of PR purpose/changes

What it says on the tin

# Jira Ticket / Github Issue #
- [X] Added the Jira Ticket to the title of the PR e.g. (DATAUP-69 Adds a PR template)

# Testing Instructions
* Details for how to test the PR: 
- [x] Tests pass in Github Actions and locally 
- [x] Changes available by spinning up a local test suite

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/truss.works/kbasetruss/development
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [n/a] I have made corresponding changes to the documentation
- [?] My changes generate no new warnings - 3k warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [n/a] Any dependent changes have been merged and published in downstream modules
- [x] I have run Black and Flake8 on changed Python Code manually or with git precommit (and the Github Actions build passes)

# Updating Version and Release Notes (if applicable)

- [n/a] [Version has been bumped](https://semver.org/) for each release
- [n/a] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: lib/execution_engine2/sdk/EE2Runjob.py [hunk 1]
```diff
@@ -338,7 +338,7 @@ def run_batch(
             self._check_workspace_permissions_list(wsids)
 
         self._add_job_requirements(params)
-        self._check_job_arguments(params)
+        self._check_job_arguments(params, has_parent_job=True)
```
Comment (MrCreosote) [inline lib/execution_engine2/sdk/EE2Runjob.py:341]: All recompiled docstring changes

File: lib/execution_engine2/execution_engine2Impl.py [hunk 2]
```diff
@@ -30,7 +30,7 @@ class execution_engine2:
     ######################################### noqa
     VERSION = "0.0.5"
     GIT_URL = "https://github.com/mrcreosote/execution_engine2.git"
-    GIT_COMMIT_HASH = "462a6110e5837e67e574e6db169a79a4d67fa8b4"
```
Comment (MrCreosote) [inline lib/execution_engine2/execution_engine2Impl.py:33]: All recompiled docstring changes in here

File: lib/execution_engine2/sdk/EE2Runjob.py [hunk 3]
```diff
@@ -360,23 +360,28 @@ def _add_job_requirements(self, jobs: List[Dict[str, Any]]):
...
+                AppInfo(job.get(_METHOD), job.get(_APP_ID)),
+                job[_JOB_REQUIREMENTS],
                 UserCreds(self.sdkmr.get_user_id(), self.sdkmr.get_token()),
-                wsid=j.get(_WORKSPACE_ID),
-                source_ws_objects=j.get(_SOURCE_WS_OBJECTS),
+                wsid=job.get(_WORKSPACE_ID),
+                source_ws_objects=job.get(_SOURCE_WS_OBJECTS),
             )
+            if has_parent_job and job.get(_PARENT_JOB_ID):
+                pre = f"Job #{i + 1}: b" if len(jobs) > 1 else "B"
```
Comment (MrCreosote) [inline lib/execution_engine2/sdk/EE2Runjob.py:378]: Slightly nicer error message if there's only one job *shrug*

File: lib/execution_engine2/sdk/EE2Runjob.py [hunk 4]
```diff
@@ -360,23 +360,28 @@ def _add_job_requirements(self, jobs: List[Dict[str, Any]]):
...
+                raise IncorrectParamsException(
+                    f"{pre}atch jobs may not specify a parent job ID"
```
Comment (ialarmedalien) [inline lib/execution_engine2/sdk/EE2Runjob.py:380]: ooooh, I like the prioritisation of correct capitalisation!

**Review by ialarmedalien — APPROVED** [review]
LGTM!

**Pull request merged**